### PR TITLE
Codebase cleanup draft

### PR DESCRIPTION
**Pull request by BrandonPacewic**
### Description

My initial draft of codebase cleanup and an auto format script. Currently using `clang-format`, it's a widely used tools as part of the `llvm-project`. 

I would like to emphasize that this is a initial draft, just looking to get some feedback. Planning on separating this pr into x different parts:
- `.clang-format` file and auto format script.
- Codebase auto format updates.
- Codebase manual format updates.
- Add format validation workflow.
- Comment updates.
- Removal of old / commented out / unreachable code sections.

I'll need do the removal of unused code as part of a peer programming project with someone that is more familiar with the project than I am.

Don't worry about all the merge conflicts, leave that for me.

### Objectives
- [x] Get feedback on current formatting settings.
- [x] Get feedback on `format_all.py` (currently only windows but I will update that later).
- [x] Get feedback on the idea of having a format validation workflow.

Is this something that we even want to do in the first place? Let me know.

Note: I am aware of some sections of code where the auto formatter did not work quite right, I'm looking into workarounds.

### CODE REVIEW

File: engine/Assets/Scripts/Importer/MirabufLive.cs [hunk 1]
```diff
@@ -1,58 +1,54 @@
...
         public Assembly MiraAssembly;
 
         public enum MirabufFileState {
-	        Valid = 1,
-	        DuplicateParts = 2
-        };
+            Valid          = 1,
+            DuplicateParts = 2
+        }
+        ;
```
Comment (KyroVibe) [inline engine/Assets/Scripts/Importer/MirabufLive.cs:51]: ?

Comment (KyroVibe) [reply engine/Assets/Scripts/Importer/MirabufLive.cs:51]: I think the semi-colon isn't supposed to be there

**Review by KyroVibe — COMMENTED** [review]
#### Format
I like the format. There are a couple places where I think the spaces before the '=' character when defining variables (especially inside methods) is kinda dump, but I can live with it.
#### format_all.py
Runs great, clear to the user whats going on, no complaints.
#### Validation Workflow
Love it. So long as it doesn't try to auto format and make commits and is just used to validate PRs, I think its a great workflow to add.

Make sure you write up documentation in the READMEs. Ideally, anyone who wants to contribute should be able to read the READMEs for setting up and contributing and do it without consulting any of us. I'll be starting a contribution guide soonish

**Comment by KyroVibe** [comment]
@BrandonPacewic Please rename the action to "Format Validation (Clang)" as I've added another format validator for the exporter.

**Comment by BrandonPacewic** [comment]
@KyroVibe This can also be used to format C++ as well if you like

**Comment by KyroVibe** [comment]
> @KyroVibe This can also be used to format C++ as well if you like

That's good, but I don't believe we have any (used) C++ code in the repository. When we do, we can extend it to that. I believe there are a couple more python scripts scattered about I may add to the Black validator but I'm not sure it's worth it.

**Comment by BrandonPacewic** [comment]
I believe that all the C++ code in this repo is for emulation. Still none of that being used at the moment?

**Comment by KyroVibe** [comment]
Ye unfortunately. If you'd like, you can implement it for that. Who knows, maybe we'll revisit it

**Comment by BrandonPacewic** [comment]
Alright

**Comment by BrandonPacewic** [comment]
Unused code removal moved into #866.

**Comment by KyroVibe** [comment]
@BrandonPacewic Can this PR be closed?

**Comment by BrandonPacewic** [comment]
> @BrandonPacewic Can this PR be closed?

Yes, the only thing that is left in this pr is the adding of braces around all `for` and `if` blocks but that is not something that I think is worth doing.

**Pull request closed without merging**